File: QUESTION_37/Ans.c
// Search an element in a matrix
#include<stdio.h>
#include <stdlib.h>
#include<stdbool.h>

bool Search(int col,int Arr[][col],int row,int no)
{
	bool flag = false;
	for(int i=0;i<row;i++)
	{
		for(int j=0;j<col;j++)
		{
			if(no == Arr[i][j])
			{
				flag = true;
				break;
			}
		}
	}
	return flag;
}

int main()
{
	
	int row = 0,col = 0;
	printf("Enter the Rows and Columns : \n");
	scanf("%d%d",&row,&col);

	int Arr[row][col];

	printf("Enter the Elements : \n");
	for(int i=0;i<row;i++)
	{
		for(int j=0;j<col;j++)
		{
			scanf("%d",&Arr[i][j]);
		}
	}
	int no = 0;
	printf("Enter the Element that you want to search : \n");
	scanf("%d",&no);

	bool bRet = Search(row,Arr,col,no);

	if(bRet == true)
		printf("Element is there\n");
	else
		printf("Element is not there\n");
	return 0;
}

File: QUESTION_49/Ans.java
/*Given a row wise sorted matrix of size RxC where R and C
  are always odd,find median of the matrix. */

import java.util.*;

class MyClass
{
	public void Median(int Arr[][])
	{
		int n = r * c;
        int Arr[] = new int[n];
        int  k = 0;
        for(int i=0;i<r;i++)
        {
            for(int j=0;j<c;j++)
            {
                Arr[k] = matrix[i][j];
                k++;
            }
        }
        
        Arrays.sort(Arr);
        int idx = n / 2;
        return Arr[idx];
    }
}

class Median_2D
{
	public static void main(String Arg[])
	{
		Scanner sobj = new Scanner(System.in);
		int Arr[][];
		System.out.println("Enter the Rows : ");
		int row = sobj.nextInt();
		System.out.println("Enter the Columns : ");
		int col = sobj.nextInt();
		Arr = new int[row][col];
		System.out.println("Enter the Elements : ");
		for(int i=0;i<row;i++)
		{
			for(int j=0;j<col;j++)
			{
				Arr[i][j] = sobj.nextInt();
			}
		}

		System.out.println("Entered Elements are : ");
		for(int i=0;i<row;i++)
		{
			for(int j=0;j<col;j++)
			{
				System.out.print(Arr[i][j]+" ");
			}
			System.out.println();
		}

		MyClass mobj = new MyClass();
		int iRet = mobj.Spiral(Arr);

	}
}
